Menus/ScenarioGUI.cpp

```diff
@@ -26,2022 +12,1051 @@
+
+	void ScenarioGUI::DrawLineToSitePoint(BITMAP *drawBitmap) const {
+		int drawColor = c_GUIColorWhite;
+		int blendAmount = 0;
+
+		for (int index = 1; index < m_LineToSitePoints.size(); index++) {
+			int lineStartX = m_LineToSitePoints.at(index - 1).GetFloorIntX();
+			int lineStartY = m_LineToSitePoints.at(index - 1).GetFloorIntY();
+			int lineEndX = m_LineToSitePoints.at(index).GetFloorIntX();
+			int lineEndY = m_LineToSitePoints.at(index).GetFloorIntY();
+
+			blendAmount = 195 + RandomNum(0, 30);
+			set_screen_blender(blendAmount, blendAmount, blendAmount, blendAmount);
+			line(drawBitmap, lineStartX, lineStartY, lineEndX, lineEndY, drawColor);
+
+			blendAmount = 30 + RandomNum(0, 50);
+			set_screen_blender(blendAmount, blendAmount, blendAmount, blendAmount);
+			line(drawBitmap, lineStartX + 1, lineStartY, lineEndX + 1, lineEndY, drawColor);
+			line(drawBitmap, lineStartX - 1, lineStartY, lineEndX - 1, lineEndY, drawColor);
+			line(drawBitmap, lineStartX, lineStartY + 1, lineEndX, lineEndY + 1, drawColor);
+			line(drawBitmap, lineStartX, lineStartY - 1, lineEndX, lineEndY - 1, drawColor);
+		}
+
+		// Draw a circle around the selected site point.
+		Vector sitePos = m_PlanetCenter + m_SelectedScene->GetLocation() + m_SelectedScene->GetLocationOffset();
+		int sitePosX = sitePos.GetFloorIntX();
+		int sitePosY = sitePos.GetFloorIntY();
+		int circleRadius = 8;
+
+		blendAmount = 195 + RandomNum(0, 30);
+		set_screen_blender(blendAmount, blendAmount, blendAmount, blendAmount);
+		circle(drawBitmap, sitePosX, sitePosY, circleRadius, drawColor);
+
+		blendAmount = 120 + RandomNum(0, 50);
+		set_screen_blender(blendAmount, blendAmount, blendAmount, blendAmount);
+		circle(drawBitmap, sitePosX, sitePosY, circleRadius - 1, drawColor);
+	}
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ScenarioGUI::Draw(BITMAP *drawBitmap) const {
+		drawing_mode(DRAW_MODE_TRANS, nullptr, 0, 0);
+		DrawSitePoints(drawBitmap);
+		if (m_SelectedScene && m_ScenarioCollectionBoxes.at(ScenarioCollections::SceneInfoBox)->GetVisible()) { DrawLineToSitePoint(drawBitmap); }
+		drawing_mode(DRAW_MODE_SOLID, nullptr, 0, 0);
+
+		AllegroScreen drawScreen(drawBitmap);
+		m_ScenarioGUIController->Draw(&drawScreen);
+
+		// Draw scene preview after GUI
+		if (m_ScenarioCollectionBoxes.at(ScenarioCollections::PlayerSetupBox)->GetVisible()) {
+			GUICollectionBox *playerSetupBox = m_ScenarioCollectionBoxes.at(ScenarioCollections::PlayerSetupBox);
+
+			// Draw the Player-Team matrix lines and disabled overlay effects.
+			int linePosY = 80;
+			for (int teamIndex = Activity::TeamOne; teamIndex < Activity::MaxTeamCount; ++teamIndex) {
+				if (m_SelectedActivity && (!m_SelectedActivity->TeamActive(teamIndex) || m_LockedCPUTeam == teamIndex)) {
+					// Apply a colored overlay on top of team rows that are not human-playable.
```

Comment:
I think we can remove these internal comments and just rely on the one at the top. I agree that having some comments explaining this stuff isn't bad, given all the complex drawing business involved in it.